app/jobs/api_key_inactive_deletion_job.rb

```diff
@@ -0,0 +1,17 @@
+require_relative 'google_analytics_queries'
+
+class ApiKeyInactiveDeletionJob < ApplicationJob
+  queue_as :default
+
+  def perform
+    active_api_keys = GoogleAnalyticsQueries.get_active_api_keys_from_ga(365)
+    api_keys_older_than_retention_period = User.where(user_type: 'api')
+                                                .where.not(email: nil)
+                                                .where("created_at <= :date", date: 1.year.ago)
+                                                .pluck(:api_key)
+    inactive_api_keys = api_keys_older_than_retention_period - active_api_keys
+    inactive_api_users = User.where(api_key: inactive_api_keys)
+    inactive_api_users.map { |u| NotifyMailer.api_key_inactive_deletion(u).deliver_later }
```

Comment:
A way to make this bit easier to test could be injecting the `NotifyMailer` dependency as part of the instantiation. Have you explored this option?